tests/c/simple.newcg.c

```diff
@@ -1,3 +1,4 @@
+// ignore-if: test YK_JIT_COMPILER != "yk"
```

Comment:
Can we just pick up `YKD_NEW_CODEGEN` here?

https://ykjit.github.io/yk/dev/runtime_config.html#ykd_new_codegen

Reply:
`YKD_NEW_CODEGEN` is like calling a road "new road" -- when you build another road should it be called "newer road"? The directory structure in ykrt/compiler gives these things names, so we should IMHO use those names and not a boolean `NEW_CODEGEN` env var.

Reply:
Why don't we push a commit to rename the existing `YKD_NEW_CODEGEN` env var and use that instead? I don't see any benefit to having two environment variables for the same purpose.

Reply:
That's probably a good idea, but the `YK_` precedent is already present in the test infrastructure, so renaming `YKD_NEW_CODEGEN` feels like a separate PR to me (and best done by someone who knows exactly what `YKD_NEW_CODEGEN` does).

Reply:
I agree. So this PR should follow precedent, and a follow up PR can do the rename.

Reply:
(I can do the rename)

Reply:
Thank you!